std/container/slist.d

```diff
@@ -589,6 +602,39 @@ Complexity: $(BIGOH n)
 
 /// ditto
     alias stableLinearRemove = linearRemove;
+
+/**
+Removes an element from the list in linear time.
+
+Complexity: $(BIGOH n)
+     */
+    void linearRemoveElement(T value)
+    {
+        enforce(_first);
+        auto n1 = findNodeByValue(_root, value);
+
+        // if node exists
+        if (n1._next)
+            n1._next = n1._next._next;
+    }
+}
+
+@safe unittest
+{
+    import std.algorithm.comparison : equal;
+
+    auto a = SList!int(-1, 1, 2, 3, 4);
+    a.linearRemoveElement(1);
+    assert(equal(a[], [-1, 2, 3, 4]));
+    a.linearRemoveElement(-1);
+    assert(equal(a[], [2, 3, 4]));
+    a.linearRemoveElement(2);
+    a.linearRemoveElement(20);
+    assert(equal(a[], [3, 4]));
+    a.linearRemoveElement(4);
+    assert(equal(a[], [3]));
+    a.linearRemoveElement(3);
+    assert(a.empty());
 }
```

Comment:
Would need a test for the behavior when one value occurs multiple times.

Reply:
Added test for this.